remotes/push.go

```diff
@@ -235,3 +245,75 @@ func pushBundleConfigDescriptor(ctx context.Context, name string, resolver remot
 	}
 	return descriptor, nil
 }
+
+func pushUnderTag(ctx context.Context, imageClient client.ImageAPIClient, image *bundle.BaseImage, targetRef reference.Named) error {
+	ref := image.Digest
+	if ref == "" {
+		ref = image.Image
+	}
+
+	if err := imageClient.ImageTag(ctx, ref, targetRef.String()); err != nil {
+		return err
+	}
+
+	err := pushTaggedImage(ctx, imageClient, targetRef)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func pushTaggedImage(ctx context.Context, imageClient client.ImageAPIClient, targetRef reference.Named) error {
+	repoInfo, err := registry.ParseRepositoryInfo(targetRef)
+	if err != nil {
+		return err
+	}
+
+	authConfig := resolveAuthConfig(repoInfo.Index)
+	encodedAuth, err := encodeAuthToBase64(authConfig)
+	if err != nil {
+		return err
+	}
+
+	reader, err := imageClient.ImagePush(ctx, targetRef.String(), types.ImagePushOptions{
+		RegistryAuth: encodedAuth,
+	})
+	if err != nil {
+		return err
+	}
+	defer reader.Close()
+	if err := jsonmessage.DisplayJSONMessagesStream(reader, ioutil.Discard, 0, false, nil); err != nil {
```

Comment:
I think we should let the user add its own iostream here so she can disable it or print it wherever she wants.
Maybe we can add it to the `WithPushImages` ?

Reply:
Or maybe we should bind it with the event producer, like for the other images mounted or copied?

Reply:
Something like this https://github.com/docker/cnab-to-oci/blob/master/remotes/fixup.go#L69 ? WDYT @eunomie ?